test/functional/test_framework/test_node.py

```diff
@@ -316,6 +333,117 @@ def assert_memory_usage_stable(self, perc_increase_allowed=0.03):
                     perc_increase_allowed * 100, before_memory_usage, after_memory_usage,
                     perc_increase_memory_usage * 100))
 
+    @contextlib.contextmanager
+    def profile_with_perf(self, profile_name):
+        """
+        Context manager that allows easy profiling of node activity using `perf`.
+
+        Perf will sample the running node and will generate profile data in the node's
+        datadir. The profile data can then be presented using `perf report` or a graphical
```

Comment:
Since this is now a user activated feature (with `--perf`, I think most of these usage notes should go in `/test/README/md`)